x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_nav.test.tsx

```diff
@@ -82,106 +82,153 @@ describe('EngineRouter', () => {
+    const wrapper = shallow(<EngineNav />);
+    expect(wrapper.isEmptyRender()).toBe(true);
   });
 
   it('does not render without an engine name', () => {
-    setMockValues({ myRole: {} });
-    (useParams as jest.Mock).mockReturnValue({ engineName: '' });
+    setMockValues({ ...values, engineName: '' });
     const wrapper = shallow(<EngineNav />);
     expect(wrapper.isEmptyRender()).toBe(true);
   });
 
-  it('renders an engine label', () => {
-    setMockValues({ myRole: {} });
-    const wrapper = mount(<EngineNav />);
+  it('renders an engine label and badges', () => {
+    setMockValues({ ...values, isSampleEngine: false, isMetaEngine: false });
+    const wrapper = shallow(<EngineNav />);
+    const label = wrapper.find('[data-test-subj="EngineLabel"]').find('.eui-textTruncate');
+
+    expect(label.text()).toEqual('SOME-ENGINE');
+    expect(wrapper.find(EuiBadge)).toHaveLength(0);
 
-    const label = wrapper.find('[data-test-subj="EngineLabel"]').last();
-    expect(label.text()).toEqual(expect.stringContaining('SOME-ENGINE'));
+    setMockValues({ ...values, isSampleEngine: true });
+    wrapper.setProps({}); // Re-render
+    expect(wrapper.find(EuiBadge).prop('children')).toEqual('SAMPLE ENGINE');
 
-    // TODO: Test sample & meta engine conditional rendering
-    expect(label.find(EuiBadge).text()).toEqual('SAMPLE ENGINE');
+    setMockValues({ ...values, isMetaEngine: true });
+    wrapper.setProps({}); // Re-render
+    expect(wrapper.find(EuiBadge).prop('children')).toEqual('META ENGINE');
   });
 
   it('renders a default engine overview link', () => {
-    setMockValues({ myRole: {} });
     const wrapper = shallow(<EngineNav />);
     expect(wrapper.find('[data-test-subj="EngineOverviewLink"]')).toHaveLength(1);
   });
 
   it('renders an analytics link', () => {
-    setMockValues({ myRole: { canViewEngineAnalytics: true } });
+    setMockValues({ ...values, myRole: { canViewEngineAnalytics: true } });
     const wrapper = shallow(<EngineNav />);
     expect(wrapper.find('[data-test-subj="EngineAnalyticsLink"]')).toHaveLength(1);
   });
 
   it('renders a documents link', () => {
-    setMockValues({ myRole: { canViewEngineDocuments: true } });
+    setMockValues({ ...values, myRole: { canViewEngineDocuments: true } });
     const wrapper = shallow(<EngineNav />);
     expect(wrapper.find('[data-test-subj="EngineDocumentsLink"]')).toHaveLength(1);
   });
 
   it('renders a schema link', () => {
-    setMockValues({ myRole: { canViewEngineSchema: true } });
+    setMockValues({ ...values, myRole: { canViewEngineSchema: true } });
     const wrapper = shallow(<EngineNav />);
     expect(wrapper.find('[data-test-subj="EngineSchemaLink"]')).toHaveLength(1);
+  });
+
+  it('renders schema nav icons', () => {
+    const myRole = { canViewEngineSchema: true };
+    const wrapper = shallow(<EngineNav />);
+
+    setMockValues({ ...values, myRole, hasUnconfirmedSchemaFields: true });
```

Comment:
I personally think different scenarios like this should be separate tests

Reply:
Ya this is a super good call, thanks for keeping me honest!

Reply:
https://github.com/elastic/kibana/pull/83138/commits/e10cc76df71ad4845d3f54e925a31bbb326f6ea9